### DISCUSSION TITLE
What does "#warning Using TIM_DIV256_CLOCK for longest timer" mean?

### DISCUSSION THREAD

**Discussion by mark-hahn** [khoih-prog/ESP8266TimerInterrupt#29] [Q&A]
I'm getting "#warning Using TIM_DIV256_CLOCK for longest timer" in my compiler output.  What does this mean?  How do I get rid of it?  I always want my compiles to be clean of all warnings.

**Comment by khoih-prog** [comment]
The warnings were there for unsophisticated users to be aware of the effect of selecting the option.

If you're a sophisticated one, you must know how and where to get rid of it, by commenting out the `#warning` lines

https://github.com/khoih-prog/ESP8266TimerInterrupt/blob/8a003893199a999b4c3ef7e56a86f306ab08937b/src/ESP8266TimerInterrupt.h#L121-L137

**Reply by mark-hahn** [reply]
Thanks for the fast response.

 I'm guess I'm unsophisticated so I don't know what it means and therefore I cannot be aware of the effect.  What does the warning mean?

**Reply by mark-hahn** [reply]
never mind.  I was confused by the word "but".  I would have thought it should be "and".

**Reply by khoih-prog** [reply]
`Using TIM_DIV1_CLOCK for shortest and most accurate timer`

As the warning says, when you select TIM_DIV1_CLOCK, you have more accurate timer, but the longest possible ISR timer will be shorter (because of the number of bits, 32/64, used to count, `uint32_t` or `uint64_t`)

On the extreme side, `Using TIM_DIV256_CLOCK for longest timer but least accurate` means when you select `TIM_DIV256_CLOCK`, you have less accurate timer, but the longest possible ISR timer will be longest.

The selection depends on your use-case. First try the best `TIM_DIV1_CLOCK`, if not long enough, reduce gradually to `USING_TIM_DIV16`, then the worst `USING_TIM_DIV256`

**Reply by mark-hahn** [reply]
Thanks.  (I wish there was a like button.)

**Reply by khoih-prog** [reply]
You're very welcome. Your words are more than enough.
